test/jasmine/tests/register_test.js

```diff
@@ -277,53 +151,119 @@ describe('the register function', function() {
+    it('should not reregister a trace module', function() {
+        Plotly.register(mockTrace1);
+        Plotly.register(mockTrace1b);
+        expect(Registry.allCategories.categories).toBe(true);
+        expect(Registry.allCategories.array).toBe(true);
+        expect(Registry.allCategories['categories b']).toBeUndefined();
+        expect(Registry.allCategories['array b']).toBeUndefined();
+        expect(Loggers.log).toHaveBeenCalled();
+    });
+});
+
+describe('Test Registry.getModule & Registry.traceIs:', function() {
+    var fakeModule = {
+        moduleType: 'trace',
+        name: 'newtype',
+        categories: ['red', 'green'],
+        calc: function() { return 42; },
+        plot: function() { return 1000000; },
+        basePlotModule: { name: 'newbaseplot' }
+    };
+
+    beforeAll(function() {
+        Registry.register(fakeModule);
+    });
+
+    beforeEach(function() {
+        spyOn(Loggers, 'log');
+    });
+
+    it('should find the module for a type', function() {
+        expect(Registry.getModule('newtype')).toBe(fakeModule);
+        expect(Registry.getModule({type: 'newtype'})).toBe(fakeModule);
+        expect(Loggers.log).not.toHaveBeenCalled();
+    });
+
+    it('should return false for types it doesn\'t know', function() {
+        expect(Registry.getModule('notatype')).toBe(false);
+        expect(Registry.getModule({type: 'notatype'})).toBe(false);
+        expect(Loggers.log).not.toHaveBeenCalled();
+    });
+
+    it('should find the categories for this type', function() {
+        expect(Registry.traceIs('newtype', 'red')).toBe(true);
+        expect(Registry.traceIs({type: 'newtype'}, 'red')).toBe(true);
+        expect(Loggers.log).not.toHaveBeenCalled();
+    });
+
+    it('should not find other real categories', function() {
+        expect(Registry.traceIs('newtype', 'cartesian')).toBe(false);
+        expect(Registry.traceIs({type: 'newtype'}, 'cartesian')).toBe(false);
+        expect(Loggers.log).not.toHaveBeenCalled();
+    });
+
+    it('should log on unrecognized trace typed', function() {
+        expect(Registry.traceIs('nada')).toBe(false);
+        expect(Loggers.log).toHaveBeenCalled();
+    });
+});
+
+describe('Test Registry.getTransformIndices & Registry.hasTransform:', function() {
+    var transformModule = {
+        moduleType: 'transform',
+        name: 'mah-transform',
+        transform: function() {}
+    };
+
+    beforeAll(function() {
+        Plotly.register(transformModule);
+    });
+
+    it('returns an empty array if no transforms present', function() {
+        expect(Registry.getTransformIndices({}, 'groupby')).toEqual([]);
+    });
+
+    it('returns an empty array if none present', function() {
+        expect(Registry.getTransformIndices({
+            transforms: [
+                {type: 'filter'},
+                {type: 'groupby'}
+            ]
+        }, 'degauss')).toEqual([]);
+    });
+
+    it('returns an empty array if none present', function() {
```

Comment:
test case description copied from above

But also, if you leave these un-nested, the descriptions should reference the method being tested.

Reply:
done in https://github.com/plotly/plotly.js/pull/2429/commits/e05f4f0db66c6797a88d62eaa67a07a7e1f5fe32